### PR TITLE
[#EVT-9] - Showroom event: create, update

### PR DESCRIPTION
**Pull request by etorras**

### CODE REVIEW

File: src/EVT/CoreDomainBundle/Repository/ShowroomRepository.php [hunk 1]
```diff
@@ -22,6 +31,13 @@ public function save($showroom)
         $this->_em->persist($entity);
         $this->_em->flush();
         $this->setId($entity->getId(), $showroom);
+
+        $eventName = Event::onCreateShowroom;
+        if ('' != $showroom->getId())
```
Comment (marco-msg-ferrari) [inline src/EVT/CoreDomainBundle/Repository/ShowroomRepository.php]: use empty()

File: src/EVT/CoreDomainBundle/Events/Event.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace EVT\CoreDomainBundle\Events;
+
+final class Event
+{
+    const onCreateShowroom = 'evt.showroom.event.on_create';
```
Comment (egulias) [inline src/EVT/CoreDomainBundle/Events/Event.php]: Maybe change to `evt.event.showroom_create`

File: src/EVT/CoreDomainBundle/Events/ShowroomEvent.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace EVT\CoreDomainBundle\Events;
+
+use EVT\CoreDomain\Provider\Showroom;
+use Symfony\Component\EventDispatcher\Event as EventDispatcher;
+use BDK\AsyncDispatcherBundle\Model\EventDispatcher\AsyncEventInterface;
+
+class ShowroomEvent extends EventDispatcher implements AsyncEventInterface
```
Comment (egulias) [inline src/EVT/CoreDomainBundle/Events/ShowroomEvent.php]: There is no need to extend from EventDispatcher

File: src/EVT/CoreDomainBundle/Events/ShowroomEvent.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public function __construct(Showroom $showroom, $name)
+    {
+        $this->showroom = $showroom;
+        $this->name = $name;
+    }
+
+    /**
+     * @param Showroom $showroom
+     */
+    public function setShowroom($showroom)
```
Comment (egulias) [inline src/EVT/CoreDomainBundle/Events/ShowroomEvent.php]: We don't need it right now.

File: src/EVT/CoreDomainBundle/Repository/ShowroomRepository.php (hunk 1 above)
Comment (egulias) [inline src/EVT/CoreDomainBundle/Repository/ShowroomRepository.php]: if you use the `if`key word you should use `{`and `}`

**Comment by egulias** [comment]
I miss the Async Driver configuration here

File: app/config/config.yml [hunk 5]
```diff
@@ -119,3 +119,18 @@ fos_user:
...
+            host:      %rabbitmq_host%
+            port:      %rabbitmq_port%
+            user:      %rabbitmq_user%
+            password:  %rabbitmq_pass%
+            vhost:     %rabbitmq_vhost%
+    producers:
+        showroom:
+            connection: default
+            exchange_options: {name: 'showroom-exchange', type: topic, durable: true}
+            queue_options:    {routing_keys: ['*.showroom'], name:'showroom-create-queue'}
```
Comment (egulias) [inline app/config/config.yml]: maybe `queue_options:    {routing_keys: ['create.showroom'], name:'showroom-create-queue'}`

File: src/EVT/CoreDomainBundle/Resources/config/services.yml [hunk 6]
```diff
@@ -57,3 +58,19 @@ services:
...
+        calls:
+            - [setRoutingKey, ['create.showroom']]
+        tags:
+            - { name: bdk.async_event_dispatcher, event: evt.event.showroom_create }
+
+    evt.showroom.async_event_driver_update:
+        class: BDK\AsyncDispatcherBundle\Model\EventDispatcher\AsyncDriver\RabbitMQDriver
+        arguments:  [@old_sound_rabbit_mq.showroom_producer, @jms_serializer]
+        calls:
+            - [setRoutingKey, ['create.showroom']]
```
Comment (egulias) [inline src/EVT/CoreDomainBundle/Resources/config/services.yml]: `[setRoutingKey, ['update.showroom']]`

**Pull request merged**